File: CS/DXReportServer_Container.Web/MyReportsResolver.cs
using System;
using System.ComponentModel.Composition;
using DevExpress.XtraReports.Service.Extensions;
using DevExpress.XtraReports.UI;

namespace DXReportServer_Container.Web {
    [Export(typeof(IReportResolver))]
    public class MyReportsResolver : IReportResolver {
        [Import]
        public ITypeResolveService TypeResolveService { get; set; }

        XtraReport IReportResolver.Resolve(string reportName, bool getParameters) {
            var type = TypeResolveService.Resolve(reportName);
            return (XtraReport)Activator.CreateInstance(type);
        }
    }
}

File: VB/DXReportServer_Container.Web/MyReportsResolver.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel.Composition
Imports DevExpress.XtraReports.Service.Extensions
Imports DevExpress.XtraReports.UI

Namespace DXReportServer_Container.Web
	<Export(GetType(IReportResolver))> _
	Public Class MyReportsResolver
		Implements IReportResolver
		Private privateTypeResolveService As ITypeResolveService
		<Import> _
		Public Property TypeResolveService() As ITypeResolveService
			Get
				Return privateTypeResolveService
			End Get
			Set(ByVal value As ITypeResolveService)
				privateTypeResolveService = value
			End Set
		End Property

		Private Function IReportResolver_Resolve(ByVal reportName As String, ByVal getParameters As Boolean) As XtraReport Implements IReportResolver.Resolve
			Dim type = TypeResolveService.Resolve(reportName)
			Return CType(Activator.CreateInstance(type), XtraReport)
		End Function
	End Class
End Namespace